tests/test_routes.py

```diff
@@ -571,6 +572,70 @@ def test_update_shopcart_item_with_none_shopcart(self):
             )
             self.assertEqual(res.status_code, status.HTTP_404_NOT_FOUND)
 
+ 
+    def test_clear_empty_shopcart(self):
+        """ It should clear the items but not delete the shopcart """
+        shopcart = self._create_an_empty_shopcart(1)[0]
+        res = self.client.put(
+            f'{BASE_URL}/{shopcart.id}/clear'
+            )
+        logging.debug(res)
+<<<<<<< HEAD
+        self.assertEqual(res.status_code, status.HTTP_200_OK) #was the PUT rqst to clear successful
+=======
+        self.assertEqual(res.status_code, status.HTTP_200_OK) # was the PUT rqst to clear successful
+>>>>>>> 19717293c8270deac12d41f136345172e0c117c5
+        logging.debug(res.status_code)
+        res = self.client.get(f'{BASE_URL}/{shopcart.id}')
+        logging.debug(res.status_code)
+        self.assertEqual(res.status_code, status.HTTP_200_OK) # was the GET rqst successful
+        data = res.get_json()
+        logging.debug(data)
+        self.assertEqual(data['id'], shopcart.id) # is the id of the shopcart we got the same as the one we created
+        self.assertEqual(data['items'], []) # is the items list empty
+                        
+
+    def test_clear_shopcart_with_items(self):
+        """ It should clear the items but not delete the shopcart """
+        shopcart = self._create_a_shopcart_with_items(3)
+        res = self.client.get(
+            f'{BASE_URL}/{shopcart.id}'
+        )
+        data = res.get_json()
+        logging.debug(data)
+        print(data)  # should have 3 items
+        res = self.client.put(
+            f'{BASE_URL}/{shopcart.id}/clear'
+            )
+        logging.debug(res)
+        self.assertEqual(res.status_code, status.HTTP_200_OK) # was the PUT rqst to clear successful
+        logging.debug(res.status_code)
+        res = self.client.get(f'{BASE_URL}/{shopcart.id}')
+        logging.debug(res.status_code)
+        self.assertEqual(res.status_code, status.HTTP_200_OK) # was the GET rqst successful
+        data = res.get_json()
+        logging.debug(data)
+        print("after: ", data) # should be empty list
+        self.assertEqual(data['id'], shopcart.id) # is the id of the shopcart we got the same as the one we created
+        self.assertEqual(data['items'], []) # is the items list empty
+
+
+    def test_clear_shopcart_not_found(self):
+        """It should return a 404 not Found response for non-existent shopcart"""
+        shopcart = self._create_an_empty_shopcart(1)[0]
+        test_id = shopcart.id + 1
+<<<<<<< HEAD
+        #item = ItemFactory(shopcart_id=test_id)
+=======
+        item = ItemFactory(shopcart_id=test_id)
```

Comment:
I think there's no need to create an item when trying to delete a non-existent shopcart